sdk/iOS/OoyalaSkinSDK/OOSkinViewController.m

```diff
@@ -390,11 +390,17 @@ - (NSString *) version {
   return OO_SKIN_VERSION;
 }
 
-- (void)queryState {
+- (void)onUIReady {
+  if (_player.currentItem) {
+    // embedcode already set, send current item information
+    NSNotification *notification = [NSNotification notificationWithName:OOOoyalaPlayerCurrentItemChangedNotification object:nil];
+    [self bridgeErrorNotification:notification];
+  }
+
   if (_player.state == OOOoyalaPlayerStateError) {
     NSNotification *notification = [NSNotification notificationWithName:OOOoyalaPlayerErrorNotification object:nil];
     [self bridgeErrorNotification:notification];
-  } else {
+  } else if (_player.state != OOOoyalaPlayerStateReady) {
```

Comment:
what happens on the overflow case?  I think if Autoplay is enabled, and react is super slow, the video might be in a playing state